sql/catalyst/src/main/scala/org/apache/spark/sql/types/DecimalType.scala

```diff
@@ -92,31 +94,47 @@ object DecimalType extends AbstractDataType {
 
   override private[sql] def simpleString: String = "decimal"
 
-  val Unlimited: DecimalType = DecimalType(None)
+  val Maximum: DecimalType = DecimalType(Some(PrecisionInfo(MAX_PRECISION, 18)))
+
+  val Default: DecimalType = DecimalType(Some(PrecisionInfo(10, 0)))
+
+  val Unlimited: DecimalType = Maximum  // backward compatibility
 
   private[sql] object Fixed {
     def unapply(t: DecimalType): Option[(Int, Int)] =
-      t.precisionInfo.map(p => (p.precision, p.scale))
+      t.precisionInfo.map(p => (p.precision, p.scale)).orElse(Some((10, 0)))
   }
 
   private[sql] object Expression {
     def unapply(e: Expression): Option[(Int, Int)] = e.dataType match {
-      case t: DecimalType => t.precisionInfo.map(p => (p.precision, p.scale))
+      case t: DecimalType => t.precisionInfo.map(p => (p.precision, p.scale)).orElse(Some((10, 0)))
       case _ => None
     }
   }
 
-  def apply(): DecimalType = Unlimited
+  def apply(): DecimalType = Default
```

Comment:
Ah damn. Maybe put a deprecate it for now? Take a look to see how many places use it. If possible, let's remove its usage.